nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -95,92 +96,95 @@ public void before() {
+        try (CapturingEventListener listener = new CapturingEventListener(COMMENT_ADDED)) {
+            Annotation annotation = createAnnotationAndAddSubscription("CommentAdded");
+            DocumentModel annotationDocumentModel = session.getDocument(new IdRef(annotation.getId()));
+            DocumentModel annotationParentDocumentModel = session.getDocument(
+                    new IdRef((String) annotationDocumentModel.getPropertyValue(COMMENT_PARENT_ID)));
+            transactionalFeature.nextTransaction();
+
+            Event expectedEvent = listener.streamCapturedEvents()
+                                          .findFirst()
+                                          .orElseThrow(() -> new AssertionError("Event wasn't fired"));
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotationParentDocumentModel,
+                    annotatedDocumentModel);
+        }
     }
 
     @Test
     public void shouldNotifyEventWhenUpdateAnnotation() {
-        // We subscribe to the update document to check that we will not be notified about the annotation updated as
-        // document (see CommentModificationVeto), only the annotation updated.
-        Annotation annotation = createAnnotationAndAddSubscription("CommentUpdated", "Modification");
-
-        captureAndVerifyAnnotationEventNotification(() -> {
+        Annotation annotation = createAnnotation(annotatedDocumentModel);
+        try (CapturingEventListener listener = new CapturingEventListener(COMMENT_UPDATED)) {
             annotation.setText("I update the annotation");
             annotationService.updateAnnotation(session, annotation.getId(), annotation);
-            return session.getDocument(new IdRef(annotation.getId()));
-        }, COMMENT_UPDATED, DOCUMENT_UPDATED);
+            DocumentModel annotationDocumentModel = session.getDocument(new IdRef(annotation.getId()));
+            DocumentModel annotationParentDocumentModel = session.getDocument(
+                    new IdRef((String) annotationDocumentModel.getPropertyValue(COMMENT_PARENT_ID)));
+            transactionalFeature.nextTransaction();
+
+            Event expectedEvent = listener.streamCapturedEvents()
+                                          .findFirst()
+                                          .orElseThrow(() -> new AssertionError("Event wasn't fired"));
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotationParentDocumentModel,
+                    annotatedDocumentModel);
+        }
     }
 
     @Test
     public void shouldNotifyEventWhenRemoveAnnotation() {
-        Annotation createdAnnotation = createAnnotationAndAddSubscription("CommentRemoved");
-        DocumentModel annotationDocModel = session.getDocument(new IdRef(createdAnnotation.getId()));
+        Annotation annotation = createAnnotation(annotatedDocumentModel);
+        DocumentModel annotationDocModel = session.getDocument(new IdRef(annotation.getId()));
```

Comment:
As this method is reworked and the variables are renamed on master, I thought it would be better to keep the same names on 10.10 even if it induces mixing a bit of Cleanup/Refacto in the commit

Same on 

- https://github.com/nuxeo/nuxeo/pull/3695/commits/db0078fd7b15fd8af33ba31fe44681623c8ec60f#diff-28554eaa66542e2d29a24a2ced0d0134R157
- https://github.com/nuxeo/nuxeo/pull/3695/commits/db0078fd7b15fd8af33ba31fe44681623c8ec60f#diff-b5ba4c2a86feb19034eeccb42bbcd8aaR136